x-pack/plugins/fleet/public/components/agent_enrollment_flyout/cloud_formation_instructions.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiButton, EuiSpacer, EuiCallOut } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+
+import { useGetSettings, useKibanaVersion, useStartServices } from '../../hooks';
+
+interface Props {
+  enrollmentAPIKey?: string;
+  cloudFormationTemplateUrl: string;
+}
+
+const createCloudFormationUrl = (
+  templateURL: string,
+  enrollmentToken: string,
+  fleetUrl: string,
+  kibanaVersion: string
+) => {
+  const cloudFormationUrl = templateURL
+    .replace('FLEET_ENROLLMENT_TOKEN', enrollmentToken)
+    .replace('FLEET_URL', fleetUrl)
+    .replace('KIBANA_VERSION', kibanaVersion);
+
+  return new URL(cloudFormationUrl).toString();
+};
+
+export const CloudFormationInstructions: React.FunctionComponent<Props> = ({
+  enrollmentAPIKey,
+  cloudFormationTemplateUrl,
+}) => {
+  const [fleetServer, setFleetServer] = useState<string | ''>();
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const core = useStartServices();
+  const { data, isLoading } = useGetSettings();
+  const { notifications } = core;
+
+  const kibanaVersion = useKibanaVersion();
+
+  // Sets Fleet Server Host as the first fleet server host available from the settings
+  // Shows an error if no fleet server host is available
+  useEffect(() => {
```

Comment:
I don't think this effect is needed. If `fleetServer` is simply a derived piece of data from `data`, it doesn't need to be stored in `state`. The only time we call `setFleetServer` is from this effect, so I think we could replace this with 

```tsx
const { data, isLoading } = useGetSettings();
const fleetServerHost = data?.item.fleet_server_hosts;
```

Highly recommend this writeup from the new React docs if you'd like to learn more about removing needless effects for deriving state: https://react.dev/learn/you-might-not-need-an-effect

Reply:
Thanks for those suggestions! `useEffect` and `useState` weren't essential here, and removing them greatly improved the code. Thanks! 🚀